src/modules/auth/services/validate-registration.auth.service.ts

```diff
@@ -0,0 +1,49 @@
+import { Inject, Injectable, UnauthorizedException } from '@nestjs/common';
+
+import { IValidateRegistrationAuthService } from '../interfaces';
+import { RegisterAuthDto } from '../domain/register.auth.dto';
+import {
+    IFindByUsernameUserService,
+    TYPES as TYPES_USER,
+  } from '../../user/interfaces';
+
+import { TYPES as TYPES_PASSPHRASE } from '../../passphrase/interfaces'
+import { TYPES as TYPES_SPACE } from '../../space/interfaces'
+import { AlreadyExistUserException } from 'src/modules/user/exceptions';
+import { ICheckPassphraseService } from 'src/modules/passphrase/interfaces/services/check.passphrase.service.interface';
+import { ICheckSpaceService } from 'src/modules/space/interfaces/services/check.space.service.interface';
+
+@Injectable()
+export class ValidateRegistrationAuthService implements IValidateRegistrationAuthService {
+  constructor(
+    @Inject(TYPES_USER.services.IFindByUsernameUserService)
+    private findUsername: IFindByUsernameUserService,
+    @Inject(TYPES_PASSPHRASE.services.ICheckPassphraseService)
+    private checkPassphrase: ICheckPassphraseService,
+    @Inject(TYPES_SPACE.services.ICheckSpaceService)
+    private checkSpace: ICheckSpaceService,
+  ) {}
+
+  async execute(data: RegisterAuthDto): Promise<boolean> {
```

Comment:
what do you think of adding a Promise.all here?

it should look like this:

```
const [passphraseTaken, user, spaceTaken] = await Promise.all([
    this.checkPassphrase.execute(data.passphrase),
    this.findUsername.execute(data.email),
    this.checkSpace.execute(data.spaceName)
  ]);    
```

then we just check each variable to check if is taken

Reply:
the main goal of this is increase the performance validating in parallel